### PR TITLE
Docs for v1 and v2

### PR DESCRIPTION
**Pull request by danielballan**

### CODE REVIEW

File: doc/source/v2/user/index.rst [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+
+.. important::
+
+   DataBroker release 1.0 includes support for old-style "v1" usage and
+   new-style "v2" usage.  This section addresses databroker's new "v2" usage.
+   It is still under development and subject to change in response to user
+   feedback.
+
+   For the stable usage "v1" usage, see :ref:`v1_index`. See
+   :ref:`transition_plan` for more information.
```
Comment (mrakitin) [inline doc/source/v2/user/index.rst:13]: I understand it's a copy of the same block, but from `v2/index.rst`. I think it's OK to keep it twice for now, but we should de-duplicate it at some point.

File: doc/source/v2/user/index.rst [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+       databroker.catalog._entries[name] = Catalog()
+
+Walkthrough
+===========
+
+Find a Catalog
+--------------
+
+When databroker is first imported, it searches for Catalogs on your system,
+typically provided by a Python package or configuartion file that you or an
```
Comment (mrakitin) [inline doc/source/v2/user/index.rst]:
```suggestion
typically provided by a Python package or configuration file that you or an
```

File: doc/source/v2/user/index.rst [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+
+      # These three lines are equivalent.
+      catalog['csx']['raw']
+      catalog['csx', 'raw']
+      catalog.csx.raw  # only works if the entry names are valid Python identifiers
+
+Look up a Run by ID
+-------------------
+
+Suppose know the unique ID of a run (a.k.a "scan") that we want to access. Note
```
Comment (mrakitin) [inline doc/source/v2/user/index.rst]:
```suggestion
Suppose you know the unique ID of a run (a.k.a "scan") that we want to access. Note
```

File: doc/source/v2/user/index.rst [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+
+.. ipython:: python
+
+   ds = run.primary.to_dask()
+   ds
+
+See the documentation on dask_.
+
+TODO: This is displaying numpy arrays, not dask. Illustrating dask here might
+requires standing up a server.
```
Comment (mrakitin) [inline doc/source/v2/user/index.rst]:
```suggestion
require standing up a server.
```

**Comment by mrakitin** [comment]
I saw a bunch of "Internal Server Errors":
```
ERROR:tornado.access:500 POST /v1/source (127.0.0.1) 5.06ms
```

Was the intake server actually started on Travis?

File: doc/source/installation.rst [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+If necessary, install it by your method of choice (apt, Homebrew, conda, etc.).
+
+Facility-Specific Distributions
+===============================
+
+*We plan for facilities that rely on databroker to provide software
+distributions that bundle databroker itself and specific Catalogs for their
+users. We will list those distributions here. For users, this will be their
+one-stop stop; they need read no futher.*
```
[the remark this answers is not in the document]
Comment (danielballan) [reply doc/source/installation.rst]: Oops, should say "one-stop ~shot~" "one-stop shop"

File: doc/source/overview.rst [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+  built on top of intake instead of databroker's own implementation. With only
+  narrow exceptions, code that worked on pre-1.0 releases should work on
+  ``databroker.v1``. This is the "default" interface in DataBroker 1.0;
+  importing ``from databroker ...`` is an alias for
+  ``from databroker.v1 ...``.
+* The module ``databroker.v2`` is a *very* thin shim around intake.
+
+The :doc:`v1/index` documentation applies to ``databroker.v0`` and
+``databroker.v1``. The :doc:`v2/index` documentation applies to
+``databroker.v2``.
```
[the remark this answers is not in the document]
Comment (danielballan) [reply doc/source/overview.rst:50]: Thanks for the suggestion!

**Comment by danielballan** [comment]
Good thought! I believe that if the server weren't started we should see connection errors. A `500` means that the client made a request that raised an unhandled Exception. In this case, I introduced a bug in 8c2ad66 that broke a critical code path, exercised by the server among other things. At the time I was focused on the docs rather than the tests or CI, so I didn't immediately notice it. I fixed the problem in a (much) later commit, 94e90ac.

File: doc/source/v2/user/index.rst (hunk 1 above)
Comment (danielballan) [reply doc/source/v2/user/index.rst:13]: Unfortunately, a limitation of rst is that you cannot include an rst file in another rst file. You can include a _txt_ file inside an rst file (which we do in just one place that I know of, whatsnew.rst) but I'm not sold that the extra ceremony is worth the de-duplication in this case.

**Comment by danielballan** [comment]
Comments addressed. Thanks again for the long phone review and the careful read, @mrakitin. Very, very appreciated!

File: doc/source/v2/user/index.rst (hunk 1 above)
Comment (mrakitin) [reply doc/source/v2/user/index.rst:13]: Agreed.

File: doc/source/installation.rst (hunk 5 above)
Comment (mrakitin) [inline doc/source/installation.rst]:
```suggestion
one-stop shop; they need read no further.*
```

File: doc/source/conf.py [hunk 7]
```diff
@@ -50,6 +50,7 @@
     'numpydoc',
     'IPython.sphinxext.ipython_directive',
     'IPython.sphinxext.ipython_console_highlighting',
+    'matplotlib.sphinxext.plot_directive',
 ]
 
 extlinks = {'issue': ('https://github.com/NSLS-II/{0}/issues/%s'.format(repo_name),
```
Comment (mrakitin) [inline doc/source/conf.py]:
```suggestion
extlinks = {'issue': ('https://github.com/bluesky/{0}/issues/%s'.format(repo_name),
```

File: doc/source/installation.rst (hunk 5 above)
Comment (mrakitin) [inline doc/source/installation.rst]:
```suggestion
one-stop shop; they need read no further.*
```

File: doc/source/conf.py (hunk 7 above)
Comment (mrakitin) [reply doc/source/conf.py]: Not sure why the GH interface tries to touch the beginning of the line -- the line starts from the very beginning without blank spaces... Feel free to make the changes on your laptop instead.

File: doc/source/installation.rst (hunk 5 above)
Comment (mrakitin) [reply doc/source/installation.rst]: For some reason it appeared twice, disregard this suggestion.

**Pull request merged**